File: benchmarks/suite/cpp/batch1.cpp
#include "../../engine/engine.h"
#include "../../common/cpp.cpp"

int main(int argc, char *argv[]) {
    const char* datasets[3];
    datasets[0] = "twitter_small_records.json";
    datasets[1] = "walmart_small_records.json";
    datasets[2] = "bestbuy_small_records.json";

    const char* parameters[3];
    parseParameters(argc, argv, parameters);
    const char* engineName = engine;
    int warmup = atoi(parameters[0]);
    int repeat = atoi(parameters[1]);
    const char* baseDir = parameters[2];
    init();

    char path[strlen(baseDir) + 150];
    execute(engineName, warmup, repeat, concat(baseDir, datasets[0], path), "TT1", queryTT1, count);
    execute(engineName, warmup, repeat, concat(baseDir, datasets[0], path), "TT2", queryTT2, count);
    execute(engineName, warmup, repeat, concat(baseDir, datasets[0], path), "TT3", queryTT3, count);
    execute(engineName, warmup, repeat, concat(baseDir, datasets[0], path), "TT4", queryTT4, count);

    execute(engineName, warmup, repeat, concat(baseDir, datasets[1], path), "WM", queryWM, count);

    execute(engineName, warmup, repeat, concat(baseDir, datasets[2], path), "BB", queryBB, count);
    
    return 0;
}

File: benchmarks/suite/python/batch1.py
import sys
sys.path.append('../../common')
sys.path.append('../../engine')
from python import execute, parseParameters
import pythonjsonpath

def main():
    twitter = "test_small_records.json"
    walmart = "walmart_small_records.json"
    bestbuy = "bestbuy_small_records.json"
    
    params = parseParameters(sys.argv)
    warmup = params["warmup"]
    repeat = params["repeat"]
    baseDir = params["datasets"]
    engineName = "pythonjsonpath"
    engine = pythonjsonpath
    engine.init()

    execute(engineName, warmup, repeat, baseDir+twitter, "TT1", engine.queryTT1, engine.count)
    execute(engineName, warmup, repeat, baseDir+twitter, "TT2", engine.queryTT2, engine.count)
    execute(engineName, warmup, repeat, baseDir+twitter, "TT3", engine.queryTT3, engine.count)
    execute(engineName, warmup, repeat, baseDir+twitter, "TT4", engine.queryTT4, engine.count)

    execute(engineName, warmup, repeat, baseDir+walmart, "WM", engine.queryWM, engine.count)

    execute(engineName, warmup, repeat, baseDir+bestbuy, "BB", engine.queryBB, engine.count)

main()